python/nix_csi/service.py

```
@@ -32,17 +36,147 @@
+        # kr8s.asyncio.get() returns an async generator, iterate with async for
+        uris = []
+        async for pod in kr8s.asyncio.get(
+            "pods", namespace=NAMESPACE, label_selector="app.kubernetes.io/name=builder"
+        ):
+            if pod.status.phase == "Running":
+                pod_name = pod.metadata.name
+                uri = f"ssh://nix@{pod_name}.{BUILDERS_SERVICE}.{NAMESPACE}.svc.cluster.local"
+                uris.append(uri)
+
+        logger.debug(f"Discovered {len(uris)} builder pods: {uris}")
+        return uris
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)